registration/include/pcl/registration/ia_fpcs.h

```diff
@@ -0,0 +1,573 @@
+        */
+      inline void
+      setMaxNormalDifference (float max_norm_diff)
+      {
+        max_norm_diff_ = max_norm_diff;
+      };
+
+      /** \return the maximum normal difference between valid point correspondences in degree. */
+      inline float
+      getMaxNormalDifference () const
+      {
+        return (max_norm_diff_);
+      };
+
+
+      /** \brief Set the maximum computation time in seconds.
+        * \param[in] max_runtime the maximum runtime of the method in seconds
+        */
+      inline void
+      setMaxComputationTime (int max_runtime)
+      {
+        max_runtime_ = max_runtime;
+      };
+
+      /** \return the maximum computation time in seconds. */
+      inline int
+      getMaxComputationTime () const
+      {
+        return (max_runtime_);
+      };
+
+
+      /** \return the fitness score of the best scored four-point match. */
+      inline float
+      getFitnessScore () const
+      {
+        return (fitness_score_);
+      };
+
+    protected:
+
+      using PCLBase <PointSource>::deinitCompute;
+      using PCLBase <PointSource>::input_;
+      using PCLBase <PointSource>::indices_;
+
+      using Registration <PointSource, PointTarget, Scalar>::reg_name_;
+      using Registration <PointSource, PointTarget, Scalar>::target_;
+      using Registration <PointSource, PointTarget, Scalar>::tree_;
+      using Registration <PointSource, PointTarget, Scalar>::correspondences_;
+      using Registration <PointSource, PointTarget, Scalar>::target_cloud_updated_;
+      using Registration <PointSource, PointTarget, Scalar>::final_transformation_;
+      using Registration <PointSource, PointTarget, Scalar>::max_iterations_;
+      using Registration <PointSource, PointTarget, Scalar>::ransac_iterations_;
+      using Registration <PointSource, PointTarget, Scalar>::transformation_estimation_;
+      using Registration <PointSource, PointTarget, Scalar>::converged_;
+
+
+      /** \brief Rigid transformation computation method.
+        * \param output the transformed input point cloud dataset using the rigid transformation found
+        * \param guess The computed transforamtion
+        */
+      virtual void
+      computeTransformation (PointCloudSource &output, const Eigen::Matrix4f& guess);
+
+
+      /** \brief Internal computation initialization. */
+      virtual bool
+      initCompute ();
+
+      /** \brief Select an approximately coplanar set of four points from the source cloud.
+        * \param[out] base_indices selected source cloud indices, further used as base (B)
+        * \param[out] ratio the two diagonal intersection ratios (r1,r2) of the base points
+        * \return
+        * * < 0 no coplanar four point sets with large enough sampling distance was found
+        * * = 0 a set of four congruent points was selected
+        */
+      int
+      selectBase (std::vector <int> &base_indices, float (&ratio)[2]);
```

Comment:
Reference-to-array is not a very common thing (at least that is my impression), therefore it might be hard to understand for some people. On the other hand, this function is not a part of public interface anyways, so I see no problem in having it this way.